kuka_rsi_driver/src/hardware_interface_eki_rsi.cpp

```diff
@@ -509,6 +564,95 @@ void KukaEkiRsiHardwareInterface::ChangeCycleTime()
   }
 }
 
+void KukaEkiRsiHardwareInterface::CopyGPIOStatesToCommands()
+{
+  auto & gpio = info_.gpios[0];
+  for (size_t i = 0; i < gpio.state_interfaces.size(); i++)
+  {
+    for (size_t j = 0; j < gpio.command_interfaces.size(); j++)
+    {
+      if (gpio.state_interfaces[i].name == gpio.command_interfaces[j].name)
+      {
+        hw_gpio_commands_[j] = hw_gpio_states_[i];
+        break;
+      }
+    }
+  }
+}
+
+kuka::external::control::kss::GPIOConfiguration KukaEkiRsiHardwareInterface::ParseGPIOConfig(
+  hardware_interface::InterfaceInfo & info)
+{
+  kuka::external::control::kss::GPIOConfiguration gpio_config;
+  gpio_config.name = info.name;
+  gpio_config.enable_limits = info.enable_limits;
+  if (info.data_type == "BOOLEAN")
+  {
+    gpio_config.value_type = kuka::external::control::GPIOValueType::BOOLEAN;
+  }
+  else if (info.data_type == "ANALOG")
+  {
+    gpio_config.value_type = kuka::external::control::GPIOValueType::ANALOG;
+  }
+  else if (info.data_type == "DIGITAL")
+  {
+    gpio_config.value_type = kuka::external::control::GPIOValueType::DIGITAL;
+  }
+  else
+  {
+    gpio_config.value_type = kuka::external::control::GPIOValueType::UNSPECIFIED;
+  }
+
+  if (!info.initial_value.empty())
+  {
+    try
+    {
+      gpio_config.initial_value = std::stod(info.initial_value);
+    }
+    catch (const std::exception & ex)
+    {
+      RCLCPP_WARN(logger_, ex.what());
```

Comment:
please extend this message to state that "initial_value is not a valid number, exception: %s"...

Reply:
for min and max too